ui-fixed and enhanced

yashgoyal0110 opened the pull request:
Implemented discussed ui changes and improvements.

1. tooltip suggestion on switch theme button
2. Sponsor button effects implemented
3. ring around footer links for smaller screens removed
4. Heading color changes on hover for projects dynamically on the basis of production/flagship/lab project

yashgoyal0110 commented:
@kasya @arkid15r I have implemented all the discussed changes, kindly review the PR.

yashgoyal0110 commented:
@arkid15r @aramattamara do we have some update on this?

aramattamara commented:
@yashgoyal0110 could you include screenshots showcasing all the changes made in this PR? It would help ensure everything is clear. Thank you!

aramattamara commented:
@yashgoyal0110 In the light theme, I noticed that the sponsor button changes to red on hover, while in the dark theme, the hover color is similar to the heart color, just slightly brighter. We discussed this with @kasya and @arkid15r and believe it would be best to fill the heart with the same color as on hover, while keeping the text unchanged.
<img width="335" alt="Screenshot 2025-01-17 at 4 54 59 PM" src="https://github.com/user-attachments/assets/c01d4f5c-e601-43c1-bff1-27d6d71a1934" />

aramattamara reviewed (changes requested):
@yashgoyal0110 In the light theme, I noticed that the sponsor button changes to red on hover, while in the dark theme, the hover color is similar to the heart color, just slightly brighter. We discussed this with @kasya and @arkid15r and believe it would be best to fill the heart with the same color as on hover, while keeping the text unchanged.
<img width="335" alt="Screenshot 2025-01-17 at 4 54 59 PM" src="https://github.com/user-attachments/assets/7a815b35-66a8-4b20-9f6f-9e91294e1473" />

yashgoyal0110 commented:
> @yashgoyal0110 could you include screenshots showcasing all the changes made in this PR? It would help ensure everything is clear. Thank you!

removed this ring around footer links which was appearing for smaller screens on click:

<img width="245" alt="Screenshot 2025-01-18 at 9 40 37 AM" src="https://github.com/user-attachments/assets/5a20f6fa-0278-4dda-9d39-4653e52125ee" />


added heading hover color effect:

<img width="1275" alt="headingHover" src="https://github.com/user-attachments/assets/98797802-9bce-40a2-8dce-9d1b880965c9" />


suggestion tooltip on them toggle button:

<img width="987" alt="suggestion tooltip on hover" src="https://github.com/user-attachments/assets/0c6e8cf8-4f44-480d-a981-914e388f2efe" />


sponsor button hover effecr:

<img width="245" alt="Screenshot 2025-01-18 at 9 45 31 AM" src="https://github.com/user-attachments/assets/0cb60e27-cadd-4e19-812c-24df9a4a7329" />

yashgoyal0110 commented:
@aramattamara hover effect fixed for sponsor button


https://github.com/user-attachments/assets/fe5157d5-ac59-4173-a9b0-16aebb8544b6

kasya commented on frontend/src/components/Card.tsx:
```diff
@@ -58,6 +58,14 @@ const Card = ({
     setShowAll(!toggleLanguagesAndTopics)
   }
 
+  const handleHeadingColor = (e, isMouseEnter) => {
```
I like all of these changes except for this one. I think having the links with different color makes it too colorful and doesn't match our overall style for the platform.

Can we revert this particular change and keep the rest?

yashgoyal0110 commented:
@kasya removed heading hover effect, ready to review

kasya reviewed (approved):
LGTM!
Let's wait for @aramattamara's review!

aramattamara reviewed (approved):
Thanks for updating the style of Sponsor button on hover. Everything else looks good to me! Ready to merge.
